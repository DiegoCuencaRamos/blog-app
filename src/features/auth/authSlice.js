import { firebase, provider }  from "../../firebase"

const inicialState = {}

// REDUCER
const authReducer = (state = inicialState, action) => {
    switch (action.type) {
        case 'auth/login':
            return {
                uid: action.payload
            }
        case 'auth/logout':
            return {}
        default:
            return state 
    }
}

// ACTION CREATORS

// 1. Login
const login = (uid) => ({
    type: 'auth/login',
    payload: uid
})

const startLogin = () => {
    return () => {
        return firebase.auth().signInWithPopup(provider);
    }
}

// 2. Logout
const logout = () => ({
    type: 'auth/logout'
})

const startLogout = () => {
    return () => {
        return firebase.auth().signOut();
    }
}

export {
    login,
    startLogin,
    logout,
    startLogout
}

export default authReducer