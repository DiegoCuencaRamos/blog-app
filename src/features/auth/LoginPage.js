import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from './authSlice'

export const LoginPage = ({ startLogin }) => (
    <section className="box-layout">
        <div className="">
            <div className="box-layout__box">
                <h1 className="box-layout__title">My blog</h1>
                <p>Lets write something awesome</p>
                <button
                    className="button"
                    onClick={startLogin}
                >
                    Login with Google
                </button>
            </div>
        </div>
    </section>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)