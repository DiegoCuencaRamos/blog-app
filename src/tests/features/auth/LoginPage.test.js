import React from 'react'
import { shallow } from 'enzyme'
import { LoginPage } from '../../../features/auth/LoginPage'

let startLogin, wrapper;

beforeEach(() =>{
    startLogin = jest.fn()
    wrapper = shallow(<LoginPage startLogin={startLogin} />)
})

// Snapshot
test('should render login page correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

// User interaction
test('should startLogin action on button clicked', () => {
    wrapper.find('button').simulate('click')
    expect(startLogin).toHaveBeenCalled()
})