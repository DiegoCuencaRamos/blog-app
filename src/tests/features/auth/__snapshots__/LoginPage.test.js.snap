// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page correctly 1`] = `
<section
  className="box-layout"
>
  <div
    className=""
  >
    <div
      className="box-layout__box"
    >
      <h1
        className="box-layout__title"
      >
        My blog
      </h1>
      <p>
        Lets write something awesome
      </p>
      <button
        className="button"
        onClick={[MockFunction]}
      >
        Login with Google
      </button>
    </div>
  </div>
</section>
`;
